### PR TITLE
ModuleInterface: consume _const keyword at var and parameter decls and keep them in textual/binary modules

### PR DESCRIPTION
**Pull request by nkcsgexi**
This is to ensure users can start adding these annotations. Type checker supports will come in later commits.

Related to pitch: https://forums.swift.org/t/pitch-compile-time-constant-values/53606

rdar://85268028

### CODE REVIEW

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test

**Comment by nkcsgexi** [comment]
@swift-ci Please smoke test OS X platform

**Comment by nkcsgexi** [comment]
@swift-ci please test

**Comment by nkcsgexi** [comment]
@swift-ci Please test OS X platform

**Comment by nkcsgexi** [comment]
@swift-ci Please Test Source Compatibility

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test

**Comment by artemcm** [comment]
Could you please document this in: [docs/ReferenceGuides/UnderscoredAttributes.md](https://github.com/apple/swift/blob/main/docs/ReferenceGuides/UnderscoredAttributes.md)

**Comment by nkcsgexi** [comment]
@swift-ci Please smoke test Linux platform

**Review by artemcm — APPROVED** [review]
This LGTM!

File: lib/AST/Expr.cpp [hunk 1]
```diff
@@ -198,6 +198,41 @@ Expr *Expr::getSemanticsProvidingExpr() {
   return this;
 }
 
+bool Expr::isSemanticallyConstExpr() const {
```
[the remark this answers is not in the document]
Comment (nkcsgexi) [reply lib/AST/Expr.cpp:201]: I found checking the constness of enum cases is the outlier. Probably we could leave that to the client-side plugin to diagnose?

**Pull request merged**